src/ducks/sep24DepositAsset.ts

```diff
@@ -0,0 +1,171 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { RootState } from "config/store";
+import { accountSelector } from "ducks/account";
+import { settingsSelector } from "ducks/settings";
+import { getNetworkConfig } from "helpers/getNetworkConfig";
+import { log } from "helpers/log";
+import {
+  sep10AuthStart,
+  sep10AuthSign,
+  sep10AuthSend,
+} from "methods/sep10Auth";
+import {
+  checkToml,
+  checkInfo,
+  interactiveDepositFlow,
+  createPopup,
+  pollDepositUntilComplete,
+} from "methods/sep24";
+import { trustAsset } from "methods/trustAsset";
+import {
+  ActionStatus,
+  Sep24DepositAssetInitialState,
+  RejectMessage,
+} from "types/types.d";
+
+export const depositAssetAction = createAsyncThunk<
+  { currentStatus: string; trustedAssetAdded?: string },
+  { assetCode: string; assetIssuer: string },
+  { rejectValue: RejectMessage; state: RootState }
+>(
+  "sep24DepositAsset/depositAssetAction",
+  async ({ assetCode, assetIssuer }, { rejectWithValue, getState }) => {
+    const { data, secretKey } = accountSelector(getState());
+    const { pubnet } = settingsSelector(getState());
+    const networkConfig = getNetworkConfig(pubnet);
+    const publicKey = data?.id;
+
+    if (!publicKey) {
+      throw new Error("Something is wrong with Account, no public key.");
+    }
+
+    log.instruction({ title: "Initiate a SEP-24 deposit" });
+
+    // TODO: get homeDomain
```

Comment:
This was refactored in https://github.com/stellar/stellar-demo-wallet/pull/125 a bit. I think it's still good to check TOML here because we show the logs (checking for SEP actions is a bit of a "magic"), which will make it more clear to the user what is going on.